cmd/zrepl/data_conn.c

```diff
@@ -154,6 +159,13 @@ uzfs_submit_writes(zvol_info_t *zinfo, zvol_io_cmd_t *zio_cmd)
 		    write_hdr->len, &metadata, is_rebuild);
 		if (rc != 0)
 			break;
+		/* Update the highest ionum used for checkpointing */
```

Comment:
rebuild IOs also will be handled by this fn. For rebuild IOs, zinfo's running_ionum should not be updated.

Reply:
ok.. noticed that ZAP won't be updated until volume is in healthy state. As we have the check for health during ZAP update, it should be fine to update running_ionum.